src/FedoraApi.php

```diff
@@ -305,4 +307,79 @@ public function getGraph(ResponseInterface $response)
         }
         return $graph;
     }
+
+    /**
+     * Creates version in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param string $timestamp Timestamp for Memento version
+     * @param string $content String or binary content
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function createVersion(
+        $uri = '',
+        $timestamp = '',
+        $content = null,
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false];
+        if ($timestamp != ''){
+            $headers['Memento-Datetime'] = $timestamp;
+        }
+        if ($content != null){
+            $options['body'] = $content;
+        }
+        $options['headers'] = $headers;
+
+        return $this->client->request(
+            'POST',
+            $timemap_uri,
+            $options
+        );
+    }
+
+    /**
+     * Gets list of versions in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function getVersions(
+        $uri = '',
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
```

Comment:
Do you need a check incase `$timemap_uri == NULL`?

Reply:
I don't know if there is a case where that header would be missing from a resource in Fedora but I think it's worth putting in a check for it. Would an acceptable thing be to do something like 
```
        if ($timemap_uri == NULL){
            throw new Exception('Timemap URI is null, cannot create version');
        }
```

Reply:
I think that would work and then we would just `try {} catch()` that in the client.

@dannylamb do you have any opinions on returning an Exception from chullo?

Reply:
I'm generally pro-exception, and would prefer throwing one instead of returning NULL or something.  There are times, in fact, when working with Chullo where I find that how it squashes exceptions makes for more work than if it just threw when 400+ or something.  I say that knowing full well that [I'm the one who made it squash exceptions](https://github.com/Islandora-CLAW/chullo/blame/dev/src/FedoraApi.php#L75)

So yeah, do it up.  I could nit and say maybe use `RuntimeException` or something, but even just plain ol' `Exception` is fine.

Reply:
changed to RuntimeException in 39bedbc